### PR TITLE
virttest.utils_test.libvirt: Update create_net_xml function.

### PR DESCRIPTION
**Pull request by rbian**
Add more sections in xml to create_net_xml; 
add a parameter define_error for negative test.

### CODE REVIEW

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -1436,7 +1436,7 @@ def file_exists():
     return diskxml.xml
 
 
-def create_net_xml(net_name, params):
+def create_net_xml(net_name, params, define_error=False):
```
Comment (will-Do) [inline virttest/utils_test/libvirt.py]: This should be a general func for test case, so deal with the result in your test case is more appropriate.

Comment (rbian) [inline virttest/utils_test/libvirt.py]: Ok, I will update it. Thanks

File: virttest/utils_test/libvirt.py [hunk 2]
```diff
@@ -1542,9 +1574,35 @@ def create_net_xml(net_name, params):
...
+                if len(pg_bandwidth_inbound) > i:
+                    pgxml.bandwidth_inbound = eval(pg_bandwidth_inbound[i])
+                if len(pg_bandwidth_outbound) > i:
+                    pgxml.bandwidth_outbound = eval(pg_bandwidth_outbound[i])
+                if len(pg_vlan) > i:
+                    pgxml.vlan_tag = eval(pg_vlan[i])
+                netxml.set_portgroup(pgxml)
         logging.debug("New network xml file: %s", netxml)
         netxml.xmltreefile.write()
-        netxml.sync()
```
Comment (yumingfei) [inline virttest/utils_test/libvirt.py]: Do you mean that it should be done by user?

**Comment by yumingfei** [comment]
Other looks fine.

File: virttest/utils_test/libvirt.py (hunk 2 above)
Comment (rbian) [inline virttest/utils_test/libvirt.py]: Yes, user will define the network and deal with exceptions.

**Comment by will-Do** [comment]
ACK
BTW, it's better to squash your commits

File: virttest/utils_test/libvirt.py [hunk 3]
```diff
@@ -1501,7 +1505,33 @@ def create_net_xml(net_name, params):
...
         forward = eval(net_forward)
         if forward:
+            net_ifs = utils_net.get_net_if(state="UP")
+            # Check forward device is valid or not,
+            # if it's not in host interface list, try to set
+            # forward device to first active interface of host
+            if (forward.has_key('mode') and forward['mode'] in
+                ['passthrough', 'private', 'bridge', 'macvtap'] and
+                forward.has_key('dev') and
+                    forward['dev'] not in net_ifs):
```
Comment (Hao-Liu) [inline virttest/utils_test/libvirt.py]: I think there is a logic error. forward device will be set if `forward` doesn't have key `mode` or `dev`, or `mode` is not valid.

File: virttest/utils_test/libvirt.py [hunk 4]
```diff
@@ -1501,7 +1505,33 @@ def create_net_xml(net_name, params):
...
+            net_ifs = utils_net.get_net_if(state="UP")
+            # The guest will use first interface of the list,
+            # check if it's valid or not
+            # if it's not in host interface list, try to set
+            # forward interface to first active interface of host
+            if interface[0]['dev'] not in net_ifs:
+                logging.warn("Forward interface %s is not a "
+                             " interface of host, reset to %s",
+                             interface[0]['dev'], net_ifs[0])
+            interface[0]['dev'] = net_ifs[0]
```
Comment (Hao-Liu) [inline virttest/utils_test/libvirt.py]: interface will alway be reset to `net_ifs[0]` here.

Comment (rbian) [inline virttest/utils_test/libvirt.py]: Good catch.

**Comment by Hao-Liu** [comment]
@rbian. I noticed there are multiple use of `eval` in this function, which mean someone can possibly run any code by just submit a malicious config file. I think we should replace them with `ast.literal_eval` or use `json` module for safety's sake.

File: virttest/utils_test/libvirt.py (hunk 3 above)
Comment (rbian) [inline virttest/utils_test/libvirt.py]: Yes, this is expected result. If user set a invalid mode or dev, user will deal with the error.

**Comment by rbian** [comment]
That's a good advice. But I don't think user will hack test script, that's meaningless.

File: virttest/utils_test/libvirt.py (hunk 3 above)
Comment (Hao-Liu) [inline virttest/utils_test/libvirt.py]: I see. So we just don't do anything if user intend to create an invalid XML. But what if user want to test against a non-existing interface with a correct mode? Shouldn't we let user decide whether they want to assign an interface automatically or not?

**Comment by Hao-Liu** [comment]
Yes, that's really a ridiculous assumption. But we should be prepared. right? :smiling_imp:

File: virttest/utils_test/libvirt.py (hunk 3 above)
Comment (rbian) [inline virttest/utils_test/libvirt.py]: Ok, maybe we need to add a parameter for this.

**Comment by rbian** [comment]
Thanks @Hao-Liu ' help. Update the code.

**Comment by Hao-Liu** [comment]
Merging this. Thanks @rbian .

**Pull request merged**